bugsnag/src/main/java/com/bugsnag/util/DeamonThreadFactory.java

```diff
@@ -0,0 +1,54 @@
+package com.bugsnag.util;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A version of {@link Executors#defaultThreadFactory()} which returns deamon threads
+ * This is to prevent applications from hanging waiting for the sessions scheduled task
+ * because deamon threads will be terminated on application shutdown
+ */
+public class DeamonThreadFactory implements ThreadFactory {
+    private static final AtomicInteger poolNumber = new AtomicInteger(1);
+    private final ThreadGroup group;
+    private final AtomicInteger threadNumber = new AtomicInteger(1);
+    private final String namePrefix;
+
+    /**
+     * Constructor
+     */
+    public DeamonThreadFactory() {
```

Comment:
There's a typo in the class name (Deamon -> Daemon)

Reply:
fixed